packages/pnpm/src/main.ts

```diff
@@ -109,6 +108,9 @@ export default async function run (inputArgv: string[]) {
     process.exitCode = 1
     return
   }
+  if (!config.useBetaCli) {
+    process.env['npm_config_argv'] = JSON.stringify(argv)
```

Comment:
Actually when I run the tests, they are still failing, still have no idea why moving this block down here would cause them to fail 😓